### PR TITLE
[MRG+1] ENH examples are now reproducible

### PR DESCRIPTION
**Pull request by NelleV**
- added a random seed in every single example that needed one.

supersedes #7382

### CODE REVIEW

**Comment by story645** [comment]
LGTM but super curious if you did this manually or with a script.

**Comment by NelleV** [comment]
I git grepped to find which examples needed it and did it manually. With vim it was quite straightforward and reasonable fast (~20minutes).

**Comment by QuLogic** [comment]
The code added after functions is one line too high; should be two blank lines after the function and one after the new code, but it's the other way around.

**Comment by NelleV** [comment]
@QuLogic I am not sure what you mean by this. pep8 is happy.

File: examples/pylab_examples/eventcollection_demo.py [hunk 1]
```diff
@@ -7,6 +7,10 @@
 from matplotlib.collections import EventCollection
 import numpy as np
 
+# Fixing random state for reproducibility
+np.random.seed(19680801)
```
Comment (QuLogic) [inline examples/pylab_examples/eventcollection_demo.py]: This one already has a seed.

**Comment by QuLogic** [comment]
Maybe pep8 became less strict about it, or maybe it only requires two blank lines before the function? I prefer to also use two blank lines after (at least for top-level functions).

**Comment by NelleV** [comment]
It only requires two blank lines before a function. It has never required blank lines after.

**Comment by QuLogic** [comment]
The seed is, in many places, one blank line away from the function and two blank lines away from the code. This makes it "seem" more related to the function when it's not. So the single blank line after the function is not bad; it's just the implication when coupled with the two blank lines below the seeding.

**Comment by QuLogic** [comment]
I think @ivanov noted some other files in `doc/pyplots` that need this as well.

**Comment by NelleV** [comment]
I'll do another PR for those, as they shouldn't be in that folder (it predates the gallery generation scripts).

**Comment by tacaswell** [comment]
:+1: I like the choice of the seed.

**Pull request merged**